index.js

```diff
@@ -1,17 +1,52 @@
 class SortedList {
-  constructor() {}
+  constructor() {
+    this.items = [];
+    this.length = 0;
+  }
 
-  add(item) {}
+  add(item) {
+    this.items.push(item);
+    this.items.sort((e1,e2) => e1 - e2)
 
-  get(pos) {}
+    this.length++;
+  }
 
-  max() {}
+  get(pos) {
+    if (pos > this.length) {
+      throw new Error('OutOfBounds');
+    }
 
-  min() {}
+    return this.items[pos];
+  }
 
-  sum() {}
+  max() {
+    if (this.length === 0) {
+      throw new Error('EmptySortedList');
+    }
 
-  avg() {}
+    return Math.max(...this.items);
```

Comment:
Perfecto!!!
Pero 3 preguntas te voy a hacer:
1º ¿Para que sirven los 3 puntos?
2º ¿Porque son necesarios para el metodo max?
3º ¿Se te ocurre alguna otra forma de hacerlo usando metodos de arrays?

Reply:
Pues según estuve leyendo, los 3 puntos en Javascript deconstruyen el array y convierte cada elemento del array en una variable individual que después envía a la función, en nuestro caso:

this.items = [5, 7, 10];

Al llamar al método Max de esta manera,

Math.max(...this.items);

Lo que le estaríamos pasando sería:

Math.max(5, 7 ,10)

Esto es lo que yo entendí, es como que extrae el valor de cada uno de los elementos del array y los devuelve como variables individuales que el método max ya puede leer y trabajar.

Yo antes de leer sobre esta solución, lo hice primero con un bucle for y después con reduce pero utilizando también Math.max, esta primera vez, sin el uso de los 3 puntos.

  const items = [5, 7, 3];

  const max = items.reduce((e1, e2) => {
    return Math.max(e1, e2);
  });

console.log(max);

Este fue el test que hice 😊